src/utils/computeSurroundingTicks.ts

```diff
@@ -0,0 +1,74 @@
+import { Token } from '@uniswap/sdk-core'
+import { TickData } from 'constants/ticks'
+import { TickMath, tickToPrice } from '@uniswap/v3-sdk'
+import JSBI from 'jsbi'
+import { TickProcessed } from 'constants/ticks'
+
+const PRICE_FIXED_DIGITS = 8
+
+// Computes the numSurroundingTicks above or below the active tick.
+export default function computeSurroundingTicks(
+  token0: Token,
+  token1: Token,
+  activeTickProcessed: TickProcessed,
+  tickIdxToInitializedTick: { [key: string]: TickData },
+  tickSpacing: number,
+  ascending: boolean
+): TickProcessed[] {
+  let previousTickProcessed: TickProcessed = {
+    ...activeTickProcessed,
+  }
+
+  // Iterate outwards (either up or down depending on direction) from the active tick,
+  // building active liquidity for every tick.
+  let processedTicks: TickProcessed[] = []
+  for (let i = 0; i < Object.keys(tickIdxToInitializedTick).length; i++) {
```

Comment:
hmm am i misunderstanding this? we're iterating through as many ticks are defined in our map, which should be only ticks that have a non-zero liquidityGross (subset of all ticks), but then we're bumping currentTickIdx by tickSpacing every time, which isn't guaranteed to get us all the way to the end unless every possible tick is "initialized"

afaict the right (naive) solution would be to just use a while loop, and wait until we hit the break, but in practice i think we simply want the return value to only contain processed ticks for which there's actually a change in liquidity, and then rely on back-filling logic for gaps, right?

Reply:
yes, you are right. 

I've implemented your second approach. Instead of relying on a `tick->{tick, liquidityNet}` map and iterating over the tick space by `TICK_SPACING` increments, I do the following

That algorithm should have a similar time complexity (?) but span the entire initialized tick space

- sort the `{tick, liquidityNet}[]` data returned by `TickLens`
- find `pivot` in sorted array where active tick is
- iterate left/right from the pivot to build active liquidity

Anything ticks beyond those should not be initialized and liquidity is 0